### DISCUSSION TITLE
Not all devices are transferred into tuya integration in HA

### DISCUSSION THREAD

**Discussion by OrsagFra** [tuya/tuya-home-assistant#548] [Q&A]
I add as new integration the new tyua integration, but in project in tuya IoT platform I have 8 devices (all which I actuallyt have) but in HA during add of tuya integration I got proposal only for 5 of them. I missed 2 Thermostat Valve and 1 Gateway. The most important for me are these Thermostat Valves.
Thank you for any hint.

**Comment by eavargasgo** [comment]
I have the same situation. In Tuya IoT platform I have 7 devices, but in HA only 5 are reflected, I am missing 2 "Smart Ceiling Fan Switch DS02F" fans. Thanks for helping me with this problem.

Device Information
Product Name: Smart Ceiling Fan Switch-DS02F
Device ID: eb0fbce741f552fb151oay
Product Category: fskg
Device Status: Online
Activated At: 2021-10-11 19:43:38

**Reply by mbround18** [reply]
fiet smart home dimmer switches too, they are visible in Tuya but not HA

**Reply by davelj** [reply]
Exactly my problem - all Fiet dimmers fail to pull

**Reply by Trukmeister** [reply]
My dimmers were missing until one core release before the latest release of core-2021.11.5

**Comment by tsykhan** [comment]
Door/window sensors and PIR detector not work too

**Reply by Trukmeister** [reply]
Not a single device which show in the IoT Platform project show in my HA.  I have checked the authentications multiple times and tried diff accounts too but no device ever shows in HA.

**Reply by atv2016** [reply]
same here, even though MCS category should work

**Reply by atv2016** [reply]
Irs working now, door sensors are imported fine, although they are working very slow, or sometimes not at all. Can’t be used to rely on currently, but it might be because of cloud integration rather then local. But at least I can use these wifi sensors now to some extent.

still I’d rather go with the official tuya then local tuya, as they don’t support door sensors, and I can work around unsupported devices by creating scenes (which it also imports).

**Comment by GHaigh** [comment]
Covers, in particular the window blind CW2GT don't come across either.  They are in the Tuya IOT devices tab just don't come across with the integration.  I am having to use the old, but perfectly fine integration that was kindly added to HACS.
